### PR TITLE
Kickflip action for skateboards

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
port of https://github.com/BeeStation/BeeStation-Hornet/pull/7507
New hud icon for skateboard while mounted, letting you "kickflip" to put the skateboard in your hand instantly.

## Why It's Good For The Game
Its really hard to pick up your skateboard, which goes against the "fast but dangerous" concept of using the skateboard and just slowing the game down as you try to target the thin sprite. This gives you an option to quickly get off, while still keeping the difficulty if you get unmounted.

## Changelog

:cl:
add: You can now kickflip on skateboards.
/:cl:

### CODE REVIEW

File: code/modules/vehicles/vehicle_actions.dm [hunk 1]
```diff
@@ -364,6 +364,41 @@
...
 
+/datum/action/vehicle/ridden/scooter/skateboard/kflip
+	name = "Kickflip"
+	desc = "Kick your board up and catch it."
+	button_icon_state = "skateboard_ollie"  // youre still going up
+	check_flags = AB_CHECK_CONSCIOUS
+
+/datum/action/vehicle/ridden/scooter/skateboard/kflip/Trigger()
+	var/obj/vehicle/ridden/scooter/skateboard/V = vehicle_target
+	var/mob/living/L = owner
```
Comment (tralezab) [inline code/modules/vehicles/vehicle_actions.dm:375]: use a more descriptive variable than `V` and `L`, please

File: code/modules/vehicles/vehicle_actions.dm [hunk 2]
```diff
@@ -364,6 +364,41 @@
 	passtable_off(rider, VEHICLE_TRAIT)
 	vehicle.pass_flags &= ~PASSTABLE
 
+/datum/action/vehicle/ridden/scooter/skateboard/kflip
+	name = "Kickflip"
+	desc = "Kick your board up and catch it."
+	button_icon_state = "skateboard_ollie"  // youre still going up
+	check_flags = AB_CHECK_CONSCIOUS
+
+/datum/action/vehicle/ridden/scooter/skateboard/kflip/Trigger()
```
Comment (tralezab) [inline code/modules/vehicles/vehicle_actions.dm:373]:
```suggestion
/datum/action/vehicle/ridden/scooter/skateboard/kflip/Trigger(trigger_flags)
```

File: code/modules/vehicles/vehicle_actions.dm [hunk 3]
```diff
@@ -364,6 +364,41 @@
...
+		L.Paralyze(50)
+		if(prob(15))
+			V.visible_message(span_userdanger("You smack against the board, hard."), \
+			span_danger("[L] misses the landing and falls on [L.p_their()] face!)"))
+			L.emote("scream")
+			L.adjustBruteLoss(10)  // thats gonna leave a mark
+			return
+		V.visible_message(span_userdanger("You fall flat onto the board!"), \
+		span_danger("[L] misses the landing and falls on [L.p_their()] face!"))
+	else
```
Comment (tralezab) [inline code/modules/vehicles/vehicle_actions.dm:390]: trailing else case. Instead, make the above if block always return, and unindent the entire else block + remove else

**Review by tralezab — CHANGES_REQUESTED** [review]

File: code/modules/vehicles/vehicle_actions.dm [hunk 4]
```diff
@@ -364,6 +364,41 @@
...
+		L.visible_message(span_notice("[L] does a sick kickflip and catches [L.p_their()] board in midair."), \
+		span_notice("You do a sick kickflip, catching the board in midair! Stylish."))
+		playsound(V, 'sound/vehicles/skateboard_ollie.ogg', 50, TRUE)
+		L.spin(4, 1)
+		animate(L, pixel_y = -6, time = 4)
+		animate(V, pixel_y = -6, time = 3)
+		V.unbuckle_mob(L)
+		addtimer(CALLBACK(V, /obj/vehicle/ridden/scooter/skateboard/.proc/pick_up_board, L), 2)  // so the board can still handle "picking it up"
```
Comment (LemonInTheDark) [inline code/modules/vehicles/vehicle_actions.dm:398]: use second defines

Comment (LemonInTheDark) [inline code/modules/vehicles/vehicle_actions.dm:398]: Why the . here? you're using an absolute it shouldn't DO anything

File: code/modules/vehicles/vehicle_actions.dm [hunk 5]
```diff
@@ -364,6 +364,41 @@
...
+			return
+		V.visible_message(span_userdanger("You fall flat onto the board!"), \
+		span_danger("[L] misses the landing and falls on [L.p_their()] face!"))
+	else
+		L.visible_message(span_notice("[L] does a sick kickflip and catches [L.p_their()] board in midair."), \
+		span_notice("You do a sick kickflip, catching the board in midair! Stylish."))
+		playsound(V, 'sound/vehicles/skateboard_ollie.ogg', 50, TRUE)
+		L.spin(4, 1)
+		animate(L, pixel_y = -6, time = 4)
+		animate(V, pixel_y = -6, time = 3)
```
Comment (LemonInTheDark) [inline code/modules/vehicles/vehicle_actions.dm:396]: doesn't the time of pickup mean that this will still be going when they're picked up?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

**Comment by LemonInTheDark** [comment]
slowing the game down?

File: code/modules/vehicles/vehicle_actions.dm (hunk 4 above)
Comment (LemonInTheDark) [reply code/modules/vehicles/vehicle_actions.dm]: For the animate too, it operates off the same numbers

Comment (etherware-novice) [reply code/modules/vehicles/vehicle_actions.dm]: well uh do you need time defines for sub-second times?

Comment (LemonInTheDark) [reply code/modules/vehicles/vehicle_actions.dm]: 0.4 SECONDS is a lot more clean then 4, yeah

**Comment by etherware-novice** [comment]
@tralezab hello, could you re-examine this pr to see if there are any more issues with it?

File: code/modules/vehicles/vehicle_actions.dm [hunk 6]
```diff
@@ -364,6 +364,42 @@
...
+			span_danger("[rider] misses the landing and falls on [rider.p_their()] face!)"))
+			rider.emote("scream")
+			rider.adjustBruteLoss(10)  // thats gonna leave a mark
+			return
+		rider.visible_message(span_userdanger("You fall flat onto the board!"), \
+		span_danger("[rider] misses the landing and falls on [rider.p_their()] face!"))
+		return
+
+	rider.visible_message(span_notice("[rider] does a sick kickflip and catches [rider.p_their()] board in midair."), \
+	span_notice("You do a sick kickflip, catching the board in midair! Stylish."))
```
Comment (Mothblocks) [inline code/modules/vehicles/vehicle_actions.dm:393]:
```suggestion
	rider.visible_message(
		span_notice("[rider] does a sick kickflip and catches [rider.p_their()] board in midair."),
		span_notice("You do a sick kickflip, catching the board in midair! Stylish."),
	)
```

Here and everywhere else--backslashes are not necessary but this style is required for readability

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/vehicles/vehicle_actions.dm [hunk 7]
```diff
@@ -364,6 +364,44 @@
...
+	var/mob/living/rider = owner
+
+	rider.adjustStaminaLoss(board.instability)
+	if (rider.getStaminaLoss() >= 100)
+		playsound(src, 'sound/effects/bang.ogg', 20, TRUE)
+		board.unbuckle_mob(rider)
+		rider.Paralyze(50)
+		if(prob(15))
+			rider.visible_message(span_userdanger("You smack against the board, hard."), \
+			span_danger("[rider] misses the landing and falls on [rider.p_their()] face!)"))
```
Comment (Mothblocks) [inline code/modules/vehicles/vehicle_actions.dm:384]:
```suggestion
			rider.visible_message(
				span_userdanger("You smack against the board, hard."),
				span_danger("[rider] misses the landing and falls on [rider.p_their()] face!)"),
			)
```

File: code/modules/vehicles/vehicle_actions.dm [hunk 8]
```diff
@@ -364,6 +364,44 @@
...
+			rider.emote("scream")
+			rider.adjustBruteLoss(10)  // thats gonna leave a mark
+			return
+		rider.visible_message(span_userdanger("You fall flat onto the board!"), \
+		span_danger("[rider] misses the landing and falls on [rider.p_their()] face!"))
```
Comment (Mothblocks) [inline code/modules/vehicles/vehicle_actions.dm:389]:
```suggestion
		rider.visible_message(
			span_userdanger("You fall flat onto the board!"),
			span_danger("[rider] misses the landing and falls on [rider.p_their()] face!"),
		)
```

File: code/modules/vehicles/vehicle_actions.dm [hunk 9]
```diff
@@ -364,6 +364,44 @@
...
+		return
+
+	rider.visible_message(
+		span_notice("[rider] does a sick kickflip and catches [rider.p_their()] board in midair."),
+		span_notice("You do a sick kickflip, catching the board in midair! Stylish."),
+	)
+	playsound(board, 'sound/vehicles/skateboard_ollie.ogg', 50, TRUE)
+	rider.spin(4, 1)
```
Comment (Mothblocks) [inline code/modules/vehicles/vehicle_actions.dm:397]: Named parameters for 4 and 1

File: code/modules/vehicles/vehicle_actions.dm [hunk 10]
```diff
@@ -364,6 +364,44 @@
...
+			return
+		rider.visible_message(span_userdanger("You fall flat onto the board!"), \
+		span_danger("[rider] misses the landing and falls on [rider.p_their()] face!"))
+		return
+
+	rider.visible_message(
+		span_notice("[rider] does a sick kickflip and catches [rider.p_their()] board in midair."),
+		span_notice("You do a sick kickflip, catching the board in midair! Stylish."),
+	)
+	playsound(board, 'sound/vehicles/skateboard_ollie.ogg', 50, TRUE)
```
Comment (Mothblocks) [inline code/modules/vehicles/vehicle_actions.dm:396]:
```suggestion
	playsound(board, 'sound/vehicles/skateboard_ollie.ogg', 50, vary = TRUE)
```

File: code/modules/vehicles/vehicle_actions.dm [hunk 11]
```diff
@@ -364,6 +364,44 @@
...
+	button_icon_state = "skateboard_ollie"  // youre still going up
+	check_flags = AB_CHECK_CONSCIOUS
+
+/datum/action/vehicle/ridden/scooter/skateboard/kflip/Trigger(trigger_flags)
+	var/obj/vehicle/ridden/scooter/skateboard/board = vehicle_target
+	var/mob/living/rider = owner
+
+	rider.adjustStaminaLoss(board.instability)
+	if (rider.getStaminaLoss() >= 100)
+		playsound(src, 'sound/effects/bang.ogg', 20, TRUE)
```
Comment (Mothblocks) [inline code/modules/vehicles/vehicle_actions.dm:379]:
```suggestion
		playsound(src, 'sound/effects/bang.ogg', 20, vary = TRUE)
```

File: code/modules/vehicles/vehicle_actions.dm [hunk 12]
```diff
@@ -364,6 +364,44 @@
 	passtable_off(rider, VEHICLE_TRAIT)
 	vehicle.pass_flags &= ~PASSTABLE
 
+/datum/action/vehicle/ridden/scooter/skateboard/kflip
```
Comment (Mothblocks) [inline code/modules/vehicles/vehicle_actions.dm:367]: Full name--`kickflip`

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/vehicles/vehicle_actions.dm (hunk 9 above)
Comment (Mothblocks) [reply code/modules/vehicles/vehicle_actions.dm:401]: This is not resolved

File: code/modules/vehicles/vehicle_actions.dm [hunk 13]
```diff
@@ -364,6 +364,48 @@
 	passtable_off(rider, VEHICLE_TRAIT)
 	vehicle.pass_flags &= ~PASSTABLE
 
+/datum/action/vehicle/ridden/scooter/skateboard/kickflip
+	name = "Kickflip"
+	desc = "Kick your board up and catch it."
+	button_icon_state = "skateboard_ollie"  // youre still going up
```
Comment (Mothblocks) [inline code/modules/vehicles/vehicle_actions.dm:370]: Useless comment

```suggestion
	button_icon_state = "skateboard_ollie"
```

**Comment by etherware-novice** [comment]
the skaterman is real!

**Pull request merged**